lib/core/presentation/widgets/movie_card.dart

```diff
@@ -7,15 +8,24 @@ class MovieCard extends StatelessWidget {
   Widget build(BuildContext context) {
     return Padding(
       padding: const EdgeInsets.symmetric(vertical: 12.0, horizontal: 4.0),
-      child: Container(
-        height: 180,
-        width: 135,
-        decoration: BoxDecoration(
-          borderRadius: BorderRadius.circular(15),
-          image: const DecorationImage(
-            fit: BoxFit.cover,
-            image: AssetImage(
-              'asset/poster.jpg',
+      child: GestureDetector(
+        onTap: () {
+          Navigator.of(context).push(
```

Comment:
You need to start working with named routes.
Use named route navigation here instead. If this is unclear to you, let me know